packages/rfw/README.md

```diff
@@ -208,7 +328,409 @@ class _ExampleState extends State<Example> {
+
+For widgets that take lists of children, the
+[`childList`](https://pub.dev/documentation/rfw/latest/rfw/DataSource/childList.html)
+method can be used. For example, this is how `Row` is defined in
+`createCoreWidgets` (see in particular the `children` line):
+
+```
+  'Row': (BuildContext context, DataSource source) {
+    return Row(
+      mainAxisAlignment: ArgumentDecoders.enumValue<MainAxisAlignment>(MainAxisAlignment.values, source, ['mainAxisAlignment']) ?? MainAxisAlignment.start,
+      mainAxisSize: ArgumentDecoders.enumValue<MainAxisSize>(MainAxisSize.values, source, ['mainAxisSize']) ?? MainAxisSize.max,
+      crossAxisAlignment: ArgumentDecoders.enumValue<CrossAxisAlignment>(CrossAxisAlignment.values, source, ['crossAxisAlignment']) ?? CrossAxisAlignment.center,
+      textDirection: ArgumentDecoders.enumValue<TextDirection>(TextDirection.values, source, ['textDirection']),
+      verticalDirection: ArgumentDecoders.enumValue<VerticalDirection>(VerticalDirection.values, source, ['verticalDirection']) ?? VerticalDirection.down,
+      textBaseline: ArgumentDecoders.enumValue<TextBaseline>(TextBaseline.values, source, ['textBaseline']),
+      children: source.childList(['children']),
+    );
+  },
+```
+
+#### `ArgumentDecoders`
+
+It is common to need to decode types that are more structured than
+merely `int`, `double`, `bool`, or `String` scalars, for example,
+enums, `Color`s, or `Paint`s.
+
+The
+[`ArgumentDecoders`](https://pub.dev/documentation/rfw/latest/rfw/ArgumentDecoders-class.html)
+namespace offers some utility functions to make the decoding of such
+values consistent.
+
+For example, the `Row` definition above has some cases of enums. To
+decode them, it uses the
+[`ArgumentDecoders.enumValue`](https://pub.dev/documentation/rfw/latest/rfw/ArgumentDecoders/enumValue.html)
+method.
+
+#### Handlers
+
+The last kind of argument that widgets can have is callbacks.
+
+Since remote widget libraries are declarative and not code, they
+cannot represent executable closures. Instead, they are represented as
+events. For example, here is how the "7" button from the
+[calculator example](https://github.com/flutter/packages/blob/main/packages/rfw/example/wasm/logic/calculator.rfwtxt)
+is represented:
+
+```
+CalculatorButton(label: "7", onPressed: event "digit" { arguments: [7] }),
+```
+
+This creates a `CalculatorButton` widget with two arguments, `label`,
+a string, and `onPressed`, an event, whose name is "digit" and whose
+arguments are a map with one key, "arguments", whose value is a list
+with one value 7.
+
+When the button is pressed, an event with the name "digit" and the
+given arguments is reported to the `RemoteWidget`'s `onEvent`
+callback. That callback takes two arguments, the event name and the
+evnet arguments.
+
+On the implementation side, in local widget libraries, arguments like
+the "onPressed" argument above must be turned into a Dart callback.
```

Comment:
I'm not quite sure how to improve the text here. It's a bit confusing because there's multiple callbacks and multiple things one might call an "event" all happening at the same time.

e.g. for `GestureDetector.onTap`, there's going to be a user-triggerd touch event that triggers a callback that was registered (by the local widgets library) with `GestureDetector` which causes a callback that was registered (by the app developer) with the `RemoteWidgets` widget for the `onEvent` property to be triggered, and that callback's arguments are the name of the event specified in the remote widget library with the arguments created by merging all the various places that wanted to report arguments.

Reply:
Yeah. I think the thing to be done is just to spell out real explicitly what all the pieces are in an example, and what kinds of objects (remote widgets, local widgets, Flutter `Widget` widgets, etc.) they are, so that the reader can put that together.

The new several paragraphs above this one in the latest revision help a lot, and I've just written some comments in the same vein regarding the two different things called `GestureDetector`.

Reply:
yeah the new paragraphs you mention are my best attempt at explaining this... let me know if you think we need even more here.

Reply:
That looks good!